### ISSUE TITLE
Error: Missing required config value at 'app.baseUrl' after fresh npx @backstage/create-app and plugin-create

### ISSUE THREAD

**Issue by dimkk** [backstage/backstage#6476]
Hello!

Followed docs to create fresh instance of backstage and to create fresh plugin.
[According to docs](https://backstage.io/docs/plugins/create-a-plugin), I can start plugin in isolation.

## Expected Behavior

Plugin start in Isolation and I can enjoy hot reload and fast iteration time

## Current Behavior

Error: Missing required config value at 'app.baseUrl' after yarn start

## Possible Solution

I tried placing app-config.yaml in plugin directory, tried using env var APP_CONFIG_app_baseUrl - still getting this error.
May be it is suggested behavior, cause I managed to run it in backstage/backstage cloned repo. If so, It would be nice to have a note in docs about it. 

## Steps to Reproduce

```js
npx @backstage/create-app
yarn create-plugin
cd plugins/...
yarn start
```

## Context

Backstage is pretty complex environment for a newcomer, I enjoy exploring it in my way, and my way is start to build something fast. So hot reloading is pretty crucial. I could use main repo, but it's too heavy and hardware intensive :<

## Your Environment

- NodeJS Version (v12):LTS 14
- Operating System and Version (e.g. Ubuntu 14.04):Ubuntu 20.04 LTS
- Browser Information: Chrome

**Comment by OrkoHunter** [comment]
Thanks for reporting @dimkk - I was able to reproduce the error. I believe one should be able to start a standalone plugin even on a newly created Backstage app.


## TL;DR of problem

I did some debugging and found out why this is happening. TL;DR `app-config.yaml` file is loaded correctly, but the Backstage app needs a `config.d.ts` to set the visibility of `app.baseUrl` to frontend. Read more about visibility of configs [here](http://backstage.io/docs/conf/defining#visibility). This is already being done in the [backstage/backstage mono-repo](https://github.com/backstage/backstage/blob/81a6c95b0ab08df35da72b3e518630cdcabafdcc/packages/core-app-api/config.d.ts#L24), that is why you didn't encounter this problem.

## Quick fix

@dimkk Quick fix to unblock yourself, add a `config.d.ts` file in `packages/app` or `packages/backend` like [this one](https://github.com/backstage/backstage/blob/81a6c95b0ab08df35da72b3e518630cdcabafdcc/packages/core-app-api/config.d.ts#L24) which increases the visibility of `app.baseUrl` to the frontend. And then update the relevant [`package.json`](https://github.com/backstage/backstage/blob/81a6c95b0ab08df35da72b3e518630cdcabafdcc/packages/core-app-api/package.json#L62-L64) to load the file, and set it as `configSchema`. You can read more [here](https://backstage.io/docs/conf/defining).

## Details

The [backstage-cli plugin:serve command](https://github.com/backstage/backstage/blob/81a6c95b0ab08df35da72b3e518630cdcabafdcc/packages/cli/src/commands/plugin/serve.ts#L28) loads all the app config files and tries to serve the bundle for the plugin. If we inspect the loaded config at this point, it looks like this

```
Loaded config from app-config.yaml
{
  schema: { process: [Function: process], serialize: [Function: serialize] },
  appConfigs: [ { data: [Object], context: 'app-config.yaml' } ],
  frontendConfig: ConfigReader {
    data: {},
    context: 'app-config.yaml',
    fallback: undefined,
    prefix: ''
  },
  frontendAppConfigs: [ { context: 'app-config.yaml', data: {} } ]
}
```

Note that the `frontendConfig.data` is empty since there is no config schema file like `config.d.ts` so far in the new Backstage app, hence all the configs are by default set to be visible to `backend` only.

Later on, the serve bundle logic is unable to get the URL where to serve the bundle at.
https://github.com/backstage/backstage/blob/81a6c95b0ab08df35da72b3e518630cdcabafdcc/packages/cli/src/lib/bundler/config.ts#L95

## Proposed fix

The skeleton used by `npx @backstage/create-app` should have a simple `config.d.ts` file in `packages/app` which sets the visibility of `app.baseUrl` to frontend.

cc @backstage/reviewers

**Comment by dimkk** [comment]
Actually worked! Thanks!

**Comment by benjdlambert** [comment]
This looks like the default behaviour now and we can close this ticket right? https://github.com/backstage/backstage/blob/4045eaab952fec349e443990330bd1ac5b2c9c18/packages/core-app-api/config.d.ts#L21-L26

**Comment by benjdlambert** [comment]
Going to close and then re-open if anyone is running into this.

**Comment by mohammad-quanit** [comment]
@benjdlambert Ive created a frontend plugin and want to use property from `app.config.yaml` file but getting the error 

> `Missing required config value at 'env.secret' in 'app-config.yaml'`

below is my `app-config.yaml`

`yaml
app:
  title: XYZ Backstage
  baseUrl: http://127.0.0.1:3000

organization:
  name: XYZ Org

backend:

  platformRegion: 'us-east-1'
  auth:
    keys:
      - secret: xxxxxxxxxxx
  baseUrl: http://127.0.0.1:7007
  listen:
    port: 7007
 
  csp:
    connect-src: ["'self'", "http:", "https:"]

  cors:
    origin: http://127.0.0.1:3000
    methods: [GET, HEAD, PATCH, POST, PUT, DELETE]
    credentials: true

  database:
    client: pg
    connection:
      host: 127.0.0.1
      port: 5432
      user: postgres
      password: postgres

env:
  secret: xxxxxxxxxxxxxxx
`

Steps I've done:
1. add config.d.ts file in plugin folder
2. added the properties of env and make it visible
3. include config.d.ts file in package.json

Goal is to read env.secret value in frontend plugin/react app, but no success.

**Comment by mohammad-quanit** [comment]
@benjdlambert Thanks for your response The above issue is fixed now. I was also trying to set visibility on AWS secrets but it wasn't working but worked on custom secrets.
